sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeHealthcareAsyncClient.java

```diff
@@ -0,0 +1,326 @@
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.core.http.rest.PagedResponseBase;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.PollResponse;
+import com.azure.core.util.polling.PollerFlux;
+import com.azure.core.util.polling.PollingContext;
+import reactor.core.publisher.Mono;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.azure.ai.textanalytics.TextAnalyticsAsyncClient.COGNITIVE_TRACING_NAMESPACE_VALUE;
+import static com.azure.ai.textanalytics.implementation.Utility.DEFAULT_POLL_DURATION;
+import static com.azure.ai.textanalytics.implementation.Utility.inputDocumentsValidation;
+import static com.azure.ai.textanalytics.implementation.Utility.parseModelId;
+import static com.azure.ai.textanalytics.implementation.Utility.parseNextLink;
+import static com.azure.ai.textanalytics.implementation.Utility.toJobState;
+import static com.azure.ai.textanalytics.implementation.Utility.toMultiLanguageInput;
+import static com.azure.ai.textanalytics.implementation.Utility.toRecognizeHealthcareEntitiesResultCollection;
+import static com.azure.ai.textanalytics.implementation.Utility.toTextAnalyticsError;
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.tracing.Tracer.AZ_TRACING_NAMESPACE_KEY;
+
+class AnalyzeHealthcareAsyncClient {
+    private final ClientLogger logger = new ClientLogger(AnalyzeHealthcareAsyncClient.class);
+    private final TextAnalyticsClientImpl service;
+
+    /**
+     * Create an {@link AnalyzeHealthcareAsyncClient} that sends requests to the Text Analytics services's healthcare
+     * LRO endpoint.
+     *
+     * @param service The proxy service used to perform REST calls.
+     */
+    AnalyzeHealthcareAsyncClient(TextAnalyticsClientImpl service) {
+        this.service = service;
+    }
+
+    PollerFlux<TextAnalyticsOperationResult, PagedFlux<HealthcareTaskResult>> beginAnalyzeHealthcare(
+        Iterable<TextDocumentInput> documents, RecognizeHealthcareEntityOptions options, Context context) {
+        try {
+            inputDocumentsValidation(documents);
+            String modelVersion = null;
+            StringIndexType stringIndexType = null;
+            if (options != null) {
+                modelVersion = options.getModelVersion();
+                if (options.getStringIndexType() != null) {
+                    stringIndexType = StringIndexType.fromString(options.getStringIndexType().toString());
+                }
+            }
+            final Boolean finalIncludeStatistics = options == null ? null : options.isIncludeStatistics();
```

Comment:
this could be simplify by managing it like you do with `modelVersion`

Reply:
the variable can be used in lambda function have to be `final` or effectively final. I choose the final approach..

Reply:
me not knowing Java :) thanks for explaining Shawn!